examples/module/mpc/train_mpc.py

```diff
@@ -0,0 +1,125 @@
+import os
+import mpc
+import time
+import torch
+import shutil
+import argparse
+import setproctitle
+import pypose as pp
+import pickle as pkl
+import torch.optim as optim
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--n_batch', type=int, default=16)
+    parser.add_argument('--n_state', type=int, default=5)
+    parser.add_argument('--n_ctrl', type=int, default=5)
+    parser.add_argument('--T', type=int, default=5)
+    parser.add_argument('--save', type=str)
+    parser.add_argument('--work', type=str, default='work')
+    parser.add_argument('--no-cuda', action='store_true')
+    parser.add_argument('--seed', type=int, default=0)
+    args = parser.parse_args()
+
+    args.cuda = not args.no_cuda and torch.cuda.is_available()
+    t = '.'.join(["{}={}".format(x, getattr(args, x))
+                  for x in ['n_batch', 'n_state', 'n_ctrl', 'T']])
+    setproctitle.setproctitle('bamos.lqr.'+t+'.{}'.format(args.seed))
+    if args.save is None:
+        args.save = os.path.join(args.work, t, str(args.seed))
+    if os.path.exists(args.save):
+        shutil.rmtree(args.save)
+    os.makedirs(args.save, exist_ok=True)
+
+    device = 'cuda' if args.cuda else 'cpu'
+    n_batch, n_state, n_ctrl, T= args.n_batch, args.n_state, args.n_ctrl, args.T
+    n_sc = n_state + n_ctrl
+
+    expert_seed = 26
+    assert expert_seed != args.seed
+    torch.manual_seed(expert_seed)
+
+    Q = torch.tile(torch.eye(n_sc), (n_batch, T, 1, 1))
+    p = torch.tile(torch.randn(n_sc), (n_batch, T, 1))
+    C = torch.eye(n_state)
+    D = torch.zeros(n_state, n_ctrl)
+    c1 = torch.zeros(n_state)
+    c2 = torch.zeros(n_state)
+    u_init = None
+
+    expert = dict(
+        Q = torch.tile(torch.eye(n_sc, device=device), (n_batch, T, 1, 1)),
+        p = torch.tile(torch.randn(n_sc, device=device), (n_batch, T, 1)),
+        A = torch.eye(n_state, device=device) \
+            + 0.2 * torch.randn(n_state, n_state, device=device),
+        B = torch.randn(n_state, n_ctrl, device=device),
+    )
+    fname = os.path.join(args.save, 'expert.pkl')
+    with open(fname, 'wb') as fi:
+        pkl.dump(expert, fi)
+
+    torch.manual_seed(args.seed)
+    A = (torch.eye(n_state, device=device) + 0.2 * torch.randn(n_state, n_state, device=device))\
+        .requires_grad_()
+    B = torch.randn(n_state, n_ctrl, device=device).requires_grad_()
+
+    fname = os.path.join(args.save, 'pypose losses.csv')
+    loss_f = open(fname, 'w')
+    loss_f.write('im_loss,mse\n')
+    loss_f.flush()
+    fname = os.path.join(args.save, 'pypose time.csv')
+    time_d = open(fname, 'w')
+    time_d.write('backward_time,overall_time\n')
+    time_d.flush()
+
+    def get_loss(x_init, _A, _B):
+        lti = pp.module.LTI(expert['A'], expert['B'], C, D, c1, c2)
```

Comment:
the expert system only need to be generate once, out side the loop. Why don't we generate that right after defining the expert dict?